build.sh

```diff
@@ -2,19 +2,44 @@
 
 set -o verbose
 
+if [ -n "${DOCKER_IMAGE}" ]; then
+
+    docker pull ${DOCKER_IMAGE}
+    docker run --env CUSTOM_BUILD_SCRIPT --env CUSTOM_TEST_SCRIPT --env SWIFT_SNAPSHOT --env MYSQL_VER -v ${TRAVIS_BUILD_DIR}:${TRAVIS_BUILD_DIR} ${DOCKER_IMAGE} /bin/bash -c "apt-get update && apt-get install -y apt-utils debconf-utils dialog git sudo lsb-release wget libxml2 && cd $TRAVIS_BUILD_DIR && ./build.sh"
 
-if [[ $TRAVIS_OS_NAME == "osx" ]]; then
-    mysql --version || { brew update && brew install mysql@5.7 && brew link mysql@5.7 --force && mysql.server start && mysql --version; }
 else
-    export DEBIAN_FRONTEND="noninteractive"
-    mysql --version || { apt-get update && apt-get install -y mysql-server libmysqlclient-dev && service mysql start && mysql --version; }
-fi
 
-mysql_upgrade -uroot || echo "No need to upgrade"
-mysql -uroot -e "CREATE USER 'swift'@'localhost' IDENTIFIED BY 'kuery';"
-mysql -uroot -e "CREATE DATABASE IF NOT EXISTS test;"
-mysql -uroot -e "GRANT ALL ON test.* TO 'swift'@'localhost';"
+    if [[ $TRAVIS_OS_NAME == "osx" ]]; then
+        if [[ $MYSQL_VER == 5 ]]; then
+            mysql --version || { brew update && brew install mysql@5.7 && brew link mysql@5.7 --force && mysql.server start && mysql --version; }
+        else
+            mysql --version || { brew update && brew install mysql && mysql.server start && mysql --version; }
+        fi
+    else
+        export DEBIAN_FRONTEND="noninteractive"
+        if [[ $MYSQL_VER == 5 ]]; then
+            mysql --version || { apt-get update && apt-get install -y pkg-config mysql-server libmysqlclient-dev && service mysql start && mysql --version; }
```

Comment:
Isn't this going to break once Ubuntu starts installing v8 by default? (or do we just fix it if and when that happens)